src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -220,7 +220,9 @@ internal async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsync(bool
                         }
 
                         // If the project has project references, we need to compute transitive origins for their packages
-                        List<LockFileTargetLibrary> projectReferences = targetsList.SelectMany(t => t.Libraries).Where(l => l.Type == LibraryType.Project).ToList();
+                        List<LockFileTargetLibrary> projectReferences = targetsList
```

Comment:
Why is the framework null here?
Is this expected or a bug? 

I feel like the PR explains the symptoms of the problem, but not the root nor the expected impact of the change.

Reply:
I haven't been able to understand exactly what causes a null framework in the assets file. Donnie was able to reproduce the error but not me, he mentioned that deleting obj helped fix this, so it makes me believe that is a wrong state of the project.

This is an example of the assets file with the problem. 
![image](https://github.com/user-attachments/assets/2767e900-b276-4e3c-b433-4f3f671cc07a)

Reply:
Another solution is to populate the framework property using the targets. For example, in the above image targets has a dictionary of frameworks that contains a list of the libraries.

```c#
 private static List<PackageReference> GetProjectPackageReferences(IList<LockFileTarget> targetsList)
        {
            var packageReferences = new List<PackageReference>();

            foreach (var target in targetsList)
            {
                var framework = target.TargetFramework;
                var projectLibraries = target.Libraries.Where(library => library.Type == "project");

                foreach (var library in projectLibraries)
                {
                    var packageReference = new PackageReference(
                                    identity: new PackageIdentity(library.Name, library.Version),
                                    targetFramework: new NuGetFramework(framework),
                                    userInstalled: false,
                                    developmentDependency: false,
                                    requireReinstallation: false,
                                    allowedVersions: new VersionRange(library.Version));
                    packageReferences.Add(packageReference);
                }
            }

            return packageReferences
                .GroupBy(p => p.PackageIdentity)
                .Select(g => g.OrderBy(p => p.TargetFramework, FrameworkSorter).First())
                .ToList();
        }

```

Reply:
I don't know, either, but since I mentioned that clearing my obj/bin folders resolved it, it could be something to do with changing a TFM and then a cache/etc not updating properly? Just guessing.
IMO, fixing the method to handle a null is a good idea even if nobody understands why it happens. Screenshots prove it happened.

Reply:
I wasn't sure before I looked at this code either. 

Unfortunately is https://github.com/NuGet/NuGet.Client/blob/2982dbfc1bac22d71234d8498af6ad43e129a49c/src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFileTargetLibrary.cs#L71 not documented super well.

@martinrrm did point out the targets section has a framework, but I tihnk those are 2 different values, current project framework, vs the framework of the selected project.

Doing a quick look-up where the LockFileTargetLibrary for projects is generated (only 1 place in the code), https://github.com/NuGet/NuGet.Client/blob/2982dbfc1bac22d71234d8498af6ad43e129a49c/src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/LockFileUtils.cs#L524-L537, it seems like there's some case in which it is not going to be there. 

How is this value being used?
It seems it's supposed to mean the project framework used for the transitive project but can't think of why that'd be relevant in this case.

Reply:
So, the PM UI packages list depends on the project framework. Right now, we [use a sorter](https://github.com/NuGet/NuGet.Client/blob/2982dbfc1bac22d71234d8498af6ad43e129a49c/src/NuGet.Core/NuGet.Frameworks/comparers/NuGetFrameworkSorter.cs#L14) to get a specific framework to [display the packages](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs#L181) (meaning that PM UI doesn't work with multi targeting).

Also, this framework is used to [calculate the transitive origins of a package](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs#L449), and this is important in order to display transitive packages, a package without a transitive origin is not displayed. 

Since the framework can be null, I suggest not considering those projects.

Reply:
I updated the code to use the same target framework that we use to get the transitive and installed packages. https://github.com/NuGet/NuGet.Client/blob/bfc13ee02b8ff6128d123c820093a4110b2d6b52/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs#L180-L202

In the above code we are also comparing the package spec target frameworks against the ones in the assets file. that help us get the current project's framework.

Reply:
There's no commit pushed.

Reply:
Mb, just pushed now